### PR TITLE
[feat] Support multiple datasets tracking for HF trainer

### PR DESCRIPTION
**Pull request by tmynn**
Added support to track multiple datasets metrics in case one uses a dictionary of datasets for evaluation and/or training

### CODE REVIEW

**Review by mihran113 — APPROVED** [review]
Looks good, but let's just bring back the single quotes as we use single quotes everywhere else in the code

File: aim/sdk/adapters/hugging_face.py [hunk 1]
```diff
@@ -108,15 +114,38 @@ def on_log(self, args, state, control,
...
 
     def close(self):
         if self._run:
             self._run.close()
             del self._run
             self._run = None
 
+    @staticmethod
+    def find_most_common_substring(names: List[str]) -> Dict[str, int]:
+        substring_counts: Dict[str, int] = {}
```
Comment (alberttorosyan) [inline aim/sdk/adapters/hugging_face.py:133]: @tmynn I'm not sure I fully understand the code below.
As we're getting only the first key of the dict, does it make sense to fill in the rest of matches? 🤔

File: aim/sdk/adapters/hugging_face.py [hunk 2]
```diff
@@ -97,6 +96,13 @@ def on_log(self, args, state, control,
                 if log_name.startswith(prefix):
                     log_name = log_name[len(prefix):]
                     context = {'subset': prefix[:-1]}
+                    if '_' in log_name:
+                        sub_dataset = AimCallback.find_most_common_substring(
```
Comment (alberttorosyan) [inline aim/sdk/adapters/hugging_face.py:100]: lets restructure this code to make it more readable.

**Review by alberttorosyan — CHANGES_REQUESTED** [review]
Overall looks good! Requested some changes before approving.

File: aim/sdk/adapters/hugging_face.py (hunk 1 above)
Comment (tmynn) [reply aim/sdk/adapters/hugging_face.py:133]: @alberttorosyan good point! Need to take the key with max value

File: aim/sdk/adapters/hugging_face.py (hunk 2 above)
Comment (tmynn) [reply aim/sdk/adapters/hugging_face.py:100]: Any recommendations on this?

**Comment by gorarakelyan** [comment]
merging this to include in the upcoming release. let's improve code health afterwards

**Review by gorarakelyan — APPROVED** [review]

**Review by alberttorosyan — APPROVED** [review]
Approving changes. Code structure change will be applied later.

**Pull request merged**